ec2/spark_ec2.py

```diff
@@ -349,6 +351,17 @@ def launch_cluster(conn, opts, cluster_name):
     if opts.identity_file is None:
         print >> stderr, "ERROR: Must provide an identity file (-i) for ssh connections."
         sys.exit(1)
+
+    if not os.path.exists(opts.identity_file):
+        print >> stderr, "ERROR: The identity file '{f}' doesn't exist.".format(f=opts.identity_file)
+        sys.exit(1)
+
+    file_mode = os.stat(opts.identity_file).st_mode
+    if not (file_mode & S_IRUSR) or \
+       not oct(file_mode)[-2:] == '00':
```

Comment:
I would put these two conditions on the same line since it's under the 100 character limit.